### PR TITLE
feat: Add a metric to track how long optimize takes

### PR DESCRIPTION
**Pull request by lynnagara**
We don't have any metrics for `snuba optimize`. This change adds a metric
which records the timing for each part that is optimized.

### CODE REVIEW

File: snuba/optimize.py [hunk 1]
```diff
@@ -112,4 +117,6 @@ def optimize_partitions(
 
         query = (query_template % args).strip()
         logger.info(f"Optimizing partition: {part.name}")
+        start = time.time()
         clickhouse.execute(query)
+        metrics.timing("optimized_part", time.time() - start, tags={"table": table})
```
Comment (MeredithAnya) [inline snuba/optimize.py:122]: is there any value in tagging the partition as well? I'm not sure if that's needed or not

Comment (fpacifici) [reply snuba/optimize.py:122]: we cannot tag partitions. Their names are dates. Which means we will be increasing the tags over time.
You can tag partition as the number of weeks from today.

**Review by fpacifici — APPROVED** [review]
Could you also please add a metric for the entire job ?

**Review by evanh — APPROVED** [review]
Tracking how many parts we optimized might also be beneficial.

**Comment by lynnagara** [comment]
I tthink entire job time and # of optimised parts can be fairly easily inferred, but I added another metric for total job time anyway in case people find it helpful. How many parts we optimised is essentially the same as a count over the `optimized_part` metric.

**Pull request merged**